02/app.js

```diff
@@ -0,0 +1,101 @@
+const formEl = document.forms[0];
+const errorsList = document.createElement('ul');
+document.body.appendChild(errorsList);
+
+formEl.addEventListener('submit', checkFieldsData)
+
+function checkFieldsData(e) {
+    e.preventDefault();
+
+    const errorsAddr =[];
+    const errors =[]
+   
+    const email = formEl["formLogin"];
+    const pass1 = formEl["formPass1"];
+    const pass2 = formEl["formPass2"];
+    const formAccept = formEl["formAccept"];
+    const requirements = {
+        mail: {
+            condition: !email.value.includes('@'),
+            errorText: 'Incorrect mail format!',
+        },
+        formAccept: {
+            condition: !formAccept.checked,
+            errorText: "You have to accept policy in order to join our community!",
+        },
+        password: {
+            condition: [
+                ((pass1.value!==pass2.value)|| (!pass1.value || !pass2.value)),
+                pass1.value.length <= 6],
+            errorText: [
+                "Passwords either don't match or field/fields are empty!",
+                "Password must be at least 7 characters long!"
+            ]
+        },
+    }
+        
+
+    checkAndPushErrors(email, requirements.mail.condition , requirements.mail.errorText);
+    checkAndPushErrors(formAccept, requirements.formAccept.condition , requirements.formAccept.errorText);
```

Comment:
Jeśli w funkcji potrzebujesz użyć łącznika tj. "AND" często to oznacza, że trzeba podzielić funkcje na 2 osobne.
W Twoim przypadku powinno to być `validate()` oraz `addError()`, czyli coś takiego:
```
if(validate(email)) {
   addError(requirements.mail.errorText)
}
```
PS. Jeśli użyjesz pętli (o tym będzie w artykule) to będzie to miało więcej sensu ;)